packages/vue-components/src/Overlay.vue

```diff
@@ -65,30 +49,20 @@ export default {
         publish('closeOverlay');
         // to prevent scrolling of the body when overlay is overscrolled
         document.body.style.overflow = 'hidden';
+        this.show = true;
+
+        this.$nextTick(() => {
+          $(this.$refs.navMenuContainer).find('a').on('click', this.toggleNavMenu);
+        });
       } else {
         document.body.style.removeProperty('overflow');
+        this.show = false;
       }
-      this.show = !this.show;
-    },
-    navMenuLoaded() {
-      this.toggleLowerNavbar();
-      $(this.$refs.navMenuContainer).find('a').on('click', () => {
-        this.toggleNavMenu();
-      });
     },
   },
   mounted() {
-    const navMenu = this.$refs.navigationMenu;
-    const buildNav = (navMenuItems) => {
-      if (!navMenuItems) { return; }
-      for (let i = 0; i < navMenuItems.childNodes.length; i += 1) {
-        navMenu.appendChild(navMenuItems.childNodes[i].cloneNode(true));
-      }
-      this.navMenuLoaded();
-    };
-
-    if (!this.hasIdentifier) {
-      buildNav(this.getNavMenuContent());
```

Comment:
Similar to the comment for page nav, we will need this to manually pull in the first site nav or the page nav if their identifiers are not present.